### PR TITLE
New Icon: Mithril (Original, Plain) 

### LINKED ISSUE #1937: [ICON REQUEST]: Mithril.js
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### About the icon

Mithril.js is a lightweight, modern client-side framework for building single-page applications (SPAs). It offers a virtual DOM for efficient rendering and a component-based architecture for modular UI construction. Additionally, Mithril provides built-in routing and XHR utilities, streamlining SPA development.

## Links
- [Mithril Website](https://mithril.js.org/)
- [Mithril Github](https://github.com/MithrilJS/mithril.js)

## Brand Logos
- [Main Logo](https://github.com/MithrilJS/mithril.js/blob/next/assets/logo.svg)

### Additional information

There does not exist any Mithril wordmark logo.
The only logos that can be created from the original Mithril SVG include (original, plain, and line). 

I recommend using an alias for original and plain instead of having to create another SVG file.

### PR DESCRIPTION
**Pull request by Yashh56**
PR name matches the format new icon: Icon name (versions separated by a comma). More details [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#overview)

PR's base is the develop branch.

Your icons are inside a folder as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#organizational-guidelines)

SVG matches the standards laid out [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#svgStandards)

A new object is added in the devicon.json file as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#-updating-the-deviconjson-)

This PR closes #1937

### CODE REVIEW

**Review by canaleal — CHANGES_REQUESTED** [review]
Excellent work overall. Once you make the two minor adjustments to the json file, the PR will be ready for approval.

Also, you don't have to, but if you feel like it, making a Line Version would be a neat extra touch!

File: devicon.json [hunk 1]
```diff
@@ -5868,6 +5868,28 @@
         "color": "#8dc63f",
         "aliases": []
     },
+    {
+        "name": "mithril",
+        "altnames": [],
+        "tags": [
```
[the remark this answers is not in the document]
Comment (Yashh56) [reply devicon.json:5874]: I will make this changes ASAP

**Comment by Yashh56** [comment]
**Thanks @canaleal can you tell me what is difference between original and line version?  And How should I convert original to line ?**

**Comment by Snailedlt** [comment]
@Yashh56 
See [SVG-Versions](https://github.com/devicons/devicon/wiki/SVG-Versions) to see what a line version is.
See this tutorial for how to make one: https://github.com/devicons/devicon/issues/1306#issuecomment-1297625294

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Seems like you forgot to remove the path from the original in the line icon. It's also scaled so that it's outside the viewbox. Please try to follow the tutorial again and see if you're able to fix it ;)
image

**Comment by Yashh56** [comment]
_mithril-original svg - Inkscape 10_26_2023 9_00_01 PM

**Comment by Yashh56** [comment]
Is this is the right way @Snailedlt

**Comment by Snailedlt** [comment]
@Yashh56 Yes that looks correct! ship it! 🚢

File: icons/mithril/mithril-line.svg [hunk 2]
```diff
@@ -1 +1 @@
...
...4.284l-.049.369.34.152a35.737 35.737 0 0 0 14.535 3.098c5.03 0 9.826-1.047 14.176-2.928l.732-.316-.603-.522c-7.032-6.075-12.07-14.382-13.95-23.832Zm41.563.347-.162.778c-1.937 9.306-6.941 17.48-13.885 23.48l-.603.521.732.317a35.656 35.656 0 0 0 14.172 2.928c4.97 0 9.709-1.02 14.023-2.858l.346-.146-.045-.373c-1.212-9.845-6.426-18.44-13.947-24.166Zm-42.156 1.42c1.985 9.046 6.787 17.005 13.445 22.984-4.04 1.644-8.427 2.615-13.05 2.616-4.734-.003-9.403-1.011-13.745-2.887 1.32-9.247 6.267-17.276 13.35-22.713zm42.732.37c6.89 5.48 11.68 13.444 12.916 22.58-4.092 1.687-8.549 2.654-13.238 2.654h-.002c-4.482.002-8.898-.93-13.045-2.616 6.567-5.9 11.332-13.722 13.37-22.619ZM63.996 82.421l-.246.139a42.13 42.13 0 0 1-20.762 5.459c-5.032 0-9.837-.923-14.322-2.538l-.744-.267.078.787c1.787 18.142 17.126 32.37 35.725 32.37 18.53 0 33.828-14.12 35.71-32.16l.083-.778-.74.254c-4.328 1.486-8.944 2.332-13.768 2.332a42.173 42.173 0 0 1-20.768-5.46zm0 1.129a43.161 43.161 0 0 0 21.014 5.47c4.65 0 9.088-.821 13.285-2.178-2.17 17.175-16.806 30.53-34.57 30.53-17.83 0-32.503-13.456-34.586-30.73 4.36 1.481 8.989 2.378 13.85 2.378a43.117 43.117 0 0 0 21.007-5.47z" color="#000" style="-inkscape-stroke:none"/></svg>
```
Comment (Snailedlt) [inline icons/mithril/mithril-line.svg:1]: Nice! The icon looks good, but the code is unoptimized. Please optimize it :)
You can do so with SVGOMG: https://github.com/devicons/devicon/issues/1279#issuecomment-1195433256

File: devicon.json [hunk 3]
```diff
@@ -5868,6 +5868,34 @@
...
+            "frontend",
+            "js"
+        ],
+        "versions": {
+            "svg": [
+                "original",
+                "line"
+            ],
+            "font": [
+                "original"
```
Comment (Snailedlt) [inline devicon.json:5888]: Line icon can also be added as a font :)
```suggestion
                "original",
                "line"
```

File: icons/mithril/mithril-original.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...922-2.36 5.093 0 9.96.934 14.492 2.571-1-10.215-6.387-19.14-14.219-24.941Zm-35.34 60.773a35.239 35.239 0 0 0 14.332 3.055c4.961 0 9.688-1.032 13.977-2.887-7.113-6.145-12.211-14.55-14.113-24.113-7.614 5.593-12.914 14.129-14.196 23.945Zm35.067-30.813a35.17 35.17 0 0 0-13.825 2.817c1.215 9.851 6.493 18.445 14.098 24.082 7.547-5.594 12.793-14.086 14.063-23.84a35.146 35.146 0 0 0-14.336-3.059Zm7.308 30.985a35.156 35.156 0 0 0 13.973 2.887c4.902 0 9.574-1.004 13.828-2.817-1.195-9.707-6.336-18.18-13.754-23.828-1.96 9.418-7.023 17.688-14.047 23.758Zm-7.308 39.824c18.277 0 33.359-13.922 35.215-31.71-4.376 1.503-9.047 2.359-13.93 2.359a42.667 42.667 0 0 1-21.012-5.524 42.63 42.63 0 0 1-21.008 5.524c-5.093 0-9.957-.934-14.492-2.567 1.762 17.89 16.883 31.918 35.227 31.918ZM42.988 10.125c-19.523 0-35.406 15.879-35.406 35.406 0 11.5 5.535 21.703 14.055 28.176 2.504-12.05 10.09-22.23 20.386-28.289.036-12.945 5.836-24.54 14.942-32.406a35.107 35.107 0 0 0-13.977-2.887Zm42.02 0a35.08 35.08 0 0 0-13.973 2.887c9.133 7.886 14.95 19.523 14.95 32.515 0 .075-.016.145-.016.215 10.117 6.156 17.527 16.328 19.91 28.317 8.793-6.45 14.535-16.82 14.535-28.528-.004-19.523-15.883-35.406-35.406-35.406Zm0 0"/></svg>
```
Comment (Snailedlt) [inline icons/mithril/mithril-original.svg:1]: Add new line at EOF
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 128 128"><path fill="#010002" d="M106.71 82.578c-.058 23.649-19.32 42.879-42.987 42.879-23.703 0-42.985-19.285-42.985-42.992 0-.07.012-.14.012-.215C8.332 74.703 0 61.086 0 45.531 0 21.824 19.285 2.54 42.988 2.54c7.633 0 14.79 2.023 21.008 5.524 6.219-3.5 13.379-5.524 21.012-5.524 23.707 0 42.988 19.29 42.988 42.992 0 15.782-8.574 29.57-21.285 37.047ZM63.997 17.102c-7.781 5.761-13.144 14.601-14.195 24.73 4.37-1.504 9.047-2.36 13.922-2.36 5.093 0 9.96.934 14.492 2.571-1-10.215-6.387-19.14-14.219-24.941Zm-35.34 60.773a35.239 35.239 0 0 0 14.332 3.055c4.961 0 9.688-1.032 13.977-2.887-7.113-6.145-12.211-14.55-14.113-24.113-7.614 5.593-12.914 14.129-14.196 23.945Zm35.067-30.813a35.17 35.17 0 0 0-13.825 2.817c1.215 9.851 6.493 18.445 14.098 24.082 7.547-5.594 12.793-14.086 14.063-23.84a35.146 35.146 0 0 0-14.336-3.059Zm7.308 30.985a35.156 35.156 0 0 0 13.973 2.887c4.902 0 9.574-1.004 13.828-2.817-1.195-9.707-6.336-18.18-13.754-23.828-1.96 9.418-7.023 17.688-14.047 23.758Zm-7.308 39.824c18.277 0 33.359-13.922 35.215-31.71-4.376 1.503-9.047 2.359-13.93 2.359a42.667 42.667 0 0 1-21.012-5.524 42.63 42.63 0 0 1-21.008 5.524c-5.093 0-9.957-.934-14.492-2.567 1.762 17.89 16.883 31.918 35.227 31.918ZM42.988 10.125c-19.523 0-35.406 15.879-35.406 35.406 0 11.5 5.535 21.703 14.055 28.176 2.504-12.05 10.09-22.23 20.386-28.289.036-12.945 5.836-24.54 14.942-32.406a35.107 35.107 0 0 0-13.977-2.887Zm42.02 0a35.08 35.08 0 0 0-13.973 2.887c9.133 7.886 14.95 19.523 14.95 32.515 0 .075-.016.145-.016.215 10.117 6.156 17.527 16.328 19.91 28.317 8.793-6.45 14.535-16.82 14.535-28.528-.004-19.523-15.883-35.406-35.406-35.406Zm0 0"/></svg>

```

**Review by Snailedlt — CHANGES_REQUESTED** [review]

**Comment by Yashh56** [comment]
**Hey @Snailedlt 👋, Please Review the Changes **

File: icons/mithril/mithril-line.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...
```
Comment (Snailedlt) [inline icons/mithril/mithril-line.svg:1]: Remove color and inkscape-stroke style attribute```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 128 128"><path d="M42.988 2.04C19.014 2.04-.5 21.552-.5 45.53c0 15.626 8.324 29.31 20.742 36.975.023 23.959 19.52 43.451 43.48 43.451 23.841 0 43.245-19.307 43.46-43.094 12.733-7.593 21.314-21.462 21.314-37.332 0-23.973-19.51-43.492-43.488-43.492-7.624 0-14.777 2.012-21.012 5.47-6.234-3.459-13.384-5.47-21.008-5.47Zm0 1c7.545 0 14.615 2 20.762 5.46l.246.139.246-.139c6.148-3.46 13.221-5.46 20.766-5.46 23.436 0 42.488 19.06 42.488 42.491 0 15.464-8.34 28.969-20.723 36.403l-.56-.561-.002 1.203c-.057 23.378-19.092 42.381-42.488 42.381-23.432 0-42.485-19.056-42.485-42.492 0-.013.012-.087.012-.215v-.281l-.24-.147C8.735 74.362.5 60.908.5 45.532.5 22.094 19.556 3.038 42.988 3.038Zm0 6.585c-19.794 0-35.906 16.108-35.906 35.906 0 11.66 5.614 22.011 14.252 28.574l.63.479.163-.775c2.474-11.908 9.973-21.97 20.15-27.96l.244-.144.002-.285c.036-12.797 5.767-24.254 14.768-32.03l.604-.52-.73-.317a35.606 35.606 0 0 0-14.177-2.928Zm42.02 0a35.578 35.578 0 0 0-14.172 2.928l-.73.316.603.522c9.028 7.795 14.775 19.294 14.775 32.136 0 .004-.015.069-.015.215v.281l.24.147c10 6.085 17.325 16.139 19.68 27.986l.152.77.633-.463c8.915-6.54 14.74-17.06 14.74-28.932-.004-19.794-16.112-35.906-35.906-35.906Zm-42.02 1c4.492-.004 8.918.93 13.073 2.623-8.784 7.887-14.41 19.224-14.522 31.908-10.045 6-17.484 15.887-20.17 27.61C13.322 66.365 8.082 56.588 8.082 45.53c0-19.256 15.654-34.906 34.906-34.906Zm42.02 0c19.252 0 34.902 15.654 34.906 34.906 0 11.283-5.456 21.245-13.795 27.631-2.559-11.661-9.777-21.556-19.637-27.654-.005-12.83-5.675-24.302-14.543-32.26 4.154-1.693 8.579-2.628 13.069-2.623ZM63.996 16.48l-.297.22c-7.89 5.84-13.328 14.808-14.394 25.081l-.08.78.74-.256c4.321-1.488 8.943-2.332 13.76-2.332 5.031 0 9.84.922 14.322 2.54l.744.268-.078-.787c-1.014-10.36-6.477-19.411-14.418-25.293Zm0 1.295c7.353 5.577 12.411 13.966 13.568 23.575-4.354-1.482-8.983-2.377-13.84-2.377-4.639 0-9.079.82-13.269 2.175 1.201-9.526 6.237-17.834 13.541-23.373zm-.271 28.788a35.67 35.67 0 0 0-14.022 2.857l-.346.146.045.373c1.233 9.991 6.586 18.708 14.297 24.424l.297.221.299-.22c7.652-5.673 12.974-14.286 14.262-24.178l.047-.37-.34-.152a35.641 35.641 0 0 0-14.54-3.102Zm-.002 1h.002c4.734 0 9.403 1.01 13.746 2.89-1.324 9.305-6.31 17.399-13.473 22.836-7.22-5.48-12.238-13.671-13.512-23.07 4.201-1.733 8.69-2.66 13.237-2.657Zm-20.534 5.5-.632.464c-7.72 5.671-13.095 14.329-14.395 24.284l-.049.369.34.152a35.737 35.737 0 0 0 14.535 3.098c5.03 0 9.826-1.047 14.176-2.928l.732-.316-.603-.522c-7.032-6.075-12.07-14.382-13.95-23.832Zm41.563.347-.162.778c-1.937 9.306-6.941 17.48-13.885 23.48l-.603.521.732.317a35.656 35.656 0 0 0 14.172 2.928c4.97 0 9.709-1.02 14.023-2.858l.346-.146-.045-.373c-1.212-9.845-6.426-18.44-13.947-24.166Zm-42.156 1.42c1.985 9.046 6.787 17.005 13.445 22.984-4.04 1.644-8.427 2.615-13.05 2.616-4.734-.003-9.403-1.011-13.745-2.887 1.32-9.247 6.267-17.276 13.35-22.713zm42.732.37c6.89 5.48 11.68 13.444 12.916 22.58-4.092 1.687-8.549 2.654-13.238 2.654h-.002c-4.482.002-8.898-.93-13.045-2.616 6.567-5.9 11.332-13.722 13.37-22.619ZM63.996 82.421l-.246.139a42.13 42.13 0 0 1-20.762 5.459c-5.032 0-9.837-.923-14.322-2.538l-.744-.267.078.787c1.787 18.142 17.126 32.37 35.725 32.37 18.53 0 33.828-14.12 35.71-32.16l.083-.778-.74.254c-4.328 1.486-8.944 2.332-13.768 2.332a42.173 42.173 0 0 1-20.768-5.46zm0 1.129a43.161 43.161 0 0 0 21.014 5.47c4.65 0 9.088-.821 13.285-2.178-2.17 17.175-16.806 30.53-34.57 30.53-17.83 0-32.503-13.456-34.586-30.73 4.36 1.481 8.989 2.378 13.85 2.378a43.117 43.117 0 0 0 21.007-5.47z"/></svg>
```

File: devicon.json [hunk 6]
```diff
@@ -5868,6 +5868,35 @@
...
+            ]
+        },
+        "color": "#010002",
+        "aliases": [
+            {
+                "base": "original",
+                "alias": "plain"
+            }
+        ]
+    },
```
Comment (Snailedlt) [inline devicon.json:5899]: Fix indentation
```suggestion
    {
        "name": "mithril",
        "altnames": [
            "mithril.js"
        ],
        "tags": [
            "javascript",
            "framework",
            "frontend",
            "js"
        ],
        "versions": {
            "svg": [
                "original",
                "line"
            ],
            "font": [
                "original",
                "line"
            ]
        },
        "color": "#010002",
        "aliases": [
            {
                "base": "original",
                "alias": "plain"
            }
        ]
    },
```

**Review by Snailedlt — DISMISSED** [review]
I made some very minor changes.
Now it all looks good!
Thanks for another great PR @Yashh56 🎉

**Comment by Yashh56** [comment]
**My First Pull Request Has Been Accepted By Hacktoberfest Team !! Thank You @Snailedlt ,@canaleal And @weh For The Help :)**

File: icons/mithril/mithril-line.svg
Comment (canaleal) [inline icons/mithril/mithril-line.svg:1]: Great work on the line mark, but there appears to be some slight overflow on the right and left side of the SVG. 

image

**Review by canaleal — CHANGES_REQUESTED** [review]
Overall great work. Just 1 final change and it's ready.

File: icons/mithril/mithril-line.svg
Comment (Snailedlt) [reply icons/mithril/mithril-line.svg:1]: How to fix this in Inkscape:
1. **Select everything:** `ctrl + a`
2. **Group it**: `ctrl + g`
3. **Align it:** `ctrl + shift + a`, then center vertically and horizontally
image
4. **Scale it:** `ctrl + shift + m`, then change the largest pixel number to `128`
image
5. **Ungroup it:** `ctrl + shift + g`

Comment (Yashh56) [reply icons/mithril/mithril-line.svg:1]: I will make this changes ASAP

Comment (Snailedlt) [reply icons/mithril/mithril-line.svg:1]: @canaleal This looks good to me now. What do you think?

**Review by Snailedlt — APPROVED** [review]
LGTM! ✔️ 
Thank you for another great contribution! 💯 🙇

**Review by canaleal — APPROVED** [review]
Perfect. looks good to me. Nice work.

**Comment by Snailedlt** [comment]
@canaleal Great work finding those little details, just remember to resolve suggestions once they are fixed ;)

**Pull request merged**